File: packages/ts-sdk/test/unit/pagination.test.ts
import { describe, expect, it } from 'vitest';
import { createPage, getNextCursor, isTreeDxPage } from '../../src/treedx/index.js';

describe('pagination helpers', () => {
  it('preserves page metadata', () => {
    const page = createPage({ items: [1], nextCursor: 'next', hasMore: true });
    expect(isTreeDxPage(page)).toBe(true);
    expect(getNextCursor(page)).toBe('next');
    expect(page.hasMore).toBe(true);
  });
});

File: packages/elixir-sdk/test/unit/pagination_test.exs
defmodule TreeDxSdk.PaginationTest do
  use ExUnit.Case, async: true

  test "page helpers preserve metadata" do
    page = TreeDxSdk.Pagination.create_page([1, 2], next_cursor: "n", has_more: true)
    assert page.items == [1, 2]
    assert TreeDxSdk.Pagination.get_next_cursor(page) == "n"
    assert page.has_more == true
    assert TreeDxSdk.Pagination.page?(page)
  end
end
